packages/devtools_app/lib/src/performance/timeline_flame_chart.dart

```diff
@@ -390,16 +437,36 @@ class TimelineFlameChartState
     setState(() {
       _selectedFrame = selectedFrame;
     });
+    await _centerSelectedFrame();
+  }
 
+  Future<void> _centerSelectedFrame() async {
     // TODO(kenz): consider using jumpTo for some of these animations to
     // improve performance.
-
     if (_selectedFrame != null) {
       // Zoom and scroll to the frame's UI event.
+      TimeRange time;
+      TimelineEvent event;
+      if (_selectedFrame.uiEventFlow == null &&
+          _selectedFrame.rasterEventFlow == null) {
+        // TODO(kenz): should we zoom to the latest available frame?
+        Notifications.of(context)
+            .push('No timeline events available for the selected frame');
+        return;
+      } else if (_selectedFrame.uiEventFlow == null) {
+        time = _selectedFrame.rasterEventFlow.time;
+        event = _selectedFrame.rasterEventFlow;
+      } else if (_selectedFrame.rasterEventFlow == null) {
```

Comment:
Also document why these checks are the way they are. I would have expected the check to be that
`uiEventFlow != null` rather than checking that `rasterEventFlow == null` as the current way would lead to a NPE if both were null for a frame but perhaps that isn't possible.